### ISSUE TITLE
Next Step Microgrant: Storing The Whole Human Genome DNA Sequence

### ISSUE THREAD

**Issue by husamia** [filecoin-project/devgrants#589]
Next Step Microgrant: Storing The Whole Human Genome DNA Sequence

### 1. What is your project? (max 100 words)

Storing the whole human genome sequence data of the 30X depth is expensive and technically challenging today. The aim of the project is to explore the current state of Filecoin, IPFS, Web3.storage and related technologies to store and maintain a copy of a single individual human and potentially all the related annotations in an sustainable way for the long term. The goal is to come up with a fully complete prototype for storing one single individual human genome if possible completely via IPFS/Filecoin. If it’s not technically possible, then the current state of the network for such, the challenges for potential future work will be presented. 

### 2. How will IPFS, Filecoin, or related technology be used for this project? (max 100 words)

I will explore the current IPFS specification to assess the practicality of storing the raw sequence data (fastq), alignment (BAM), variants (VCF) and related meta data associated with how the data was generated and processed in current standard workflows. The file formats, and size limitations be explored. The Filecoin network will be explored to assess the cost and explore potential economic model for long term storage. Related technologies such as Web3.storage and other will be explored for feasibility to be used in such prototype. 
 
### 3. How will you improve your project with this grant? What steps will you take to meet this objective? (max 200 words)

I will sequence my own whole human genome using a direct to consumer service, download the raw sequence, and all the related data such as BAM and VCF. Then, I will explore the current IPFS/Filecoin tools and services available to store it and pay for the cost associated with storage. I will seek and pay for technical consultation from the Filecoin community to develop a working prototype needed to manipulate, transfer, and store the data completely on IPFS/Filecoin.
 
### 4. Is this project open source?
Yes [CC-BY-SA 3.0](https://ipfs.io/ipfs/QmVreNvKsQmQZ83T86cWSjPu2vR3yZHGPm5jnxFuunEB9u) 

### 5. Do you agree to share grant reports upon request, including a final grant report at the end of the three month period?
Yes, I will share progress reports of my research and experience during the prototype building stage. The technical challenges of IPFS, Filecoin, Web3.storage and all explored technologies will be presented. 

### 6. Does your proposal comply with our Community Code of Conduct?
Yes

### 7. Links and submissions
This work has not been presented anywhere. It's my own idea. 
  
Once the project is approved, I will start seeking technical sponsorship from Filecoin community. 

I learned about the microgrant program from the Filecoin Orbit 2022 Program.

**Comment by jenks-guo-filecoin** [comment]
I am very interested in this project. 🧬

**Comment by tippi-fifestarr** [comment]
How can I help?  I'm thinking I can see myself somewhere in this sentence:

> "I will seek and pay for technical consultation from the Filecoin community to develop a working prototype needed to manipulate, transfer, and store the data completely on IPFS/Filecoin."

Let me know if there's other steps I can participate in

**Comment by ErinOCon** [comment]
Hi @husamia, This microgrant has been approved! Can you provide an email address to discuss payment details?

**Comment by husamia** [comment]
> Hi @husamia, This microgrant has been approved! Can you provide an email address to discuss payment details?
This is my email address 
ammar.husami@gmail.com

**Comment by ErinOCon** [comment]
Thank you!
